src/dbnode/client/host_queue.go

```diff
@@ -515,8 +526,8 @@ func (q *queue) asyncTaggedWrite(
 
 		// NB(r): Defer is slow in the hot path unfortunately
 		cleanup := func() {
-			q.writeTaggedBatchRawRequestPool.Put(req)
 			q.writeTaggedBatchRawRequestElementArrayPool.Put(elems)
+			q.writeTaggedBatchRawRequestPool.Put(req)
```

Comment:
Curious - does this ordering actually contribute to causing the bug or is this just making it alphabetical?